pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+
+package nsmconnect;
+
+import "github.com/ligato/networkservicemesh/netmesh/model/netmesh/netmesh.proto";
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string name = 1;
+    string namespace = 2;
+    Label labels = 3;
+}
+
+message KernelInterface {
+   Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message VhostUserInterface{
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message MemInterface {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message SRIOV {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+enum InterfaceType {
+    not_specified_type = 0;
+    kernel_interface = 1;
+    vhost_user = 2;
+    memif = 3;
+    sriov = 4;
+  }
+
+enum InterfacePreference {
+    not_specified_pref = 0;
+    first = 1;
+    second = 2;
+    third = 3;
+    forth = 4;
+    fifth = 5;
+}
+
+message MechanismType {
+    InterfaceType type = 1;
+    InterfacePreference preference = 2;
+  }
+
+message AffinityPreference {
```

Comment:
What kind of values would go into this affinity preference? Would it be better off as a label in the metadata?

Reply:
I would agree, but @edwarnicke showed it on slide 21 as a separate item,  if label from metadata is sufficient, I am ok to remove it.